### DISCUSSION TITLE
Variables accessible for loops

### DISCUSSION THREAD

**Discussion by SebBerner** [ansys/pyaedt#4029] [Q&A]
For my workflow I would like to generate multiple traces, where the number of traces is controlled by a (design-) variable. My approach was to add a design or a project variable and use this in a for loop. Unfortunately this doesn't seem to work even though my variable works. Is there any workaround or it this a potantial issue in pyaedt/EDB API?

`edb = Edb("my_project.aedb", edbversion="2022.2") 
edb.stackup.add_layer(layer_name="Top", material="copper", thickness="35um", layer_type="signal")

edb["$_points"] = 2

edb.add_design_variable(variable_name = "Length", variable_value = 4, is_parameter=False)

for i in range (0,int(edb["$_points"].value)):   
    edb.modeler.create_trace(layer_name='Top', width = 0.1, path_list=[[str(i)+'*$_points',0],[str(i)+'*$_points','Length']] )

edb.save_edb()
edb.close_edb()`

Thanks,
Seb

**Comment by Samuelopez-ansys** [comment]
@svandenb-dev @ring630

**Comment by hui-zhou-a** [comment]
@SebBerner below line doesn't make much sense to me. Can you draw the expected design manually? I can better understand what you want to achieve.

`
edb.modeler.create_trace(layer_name='Top', width = 0.1, path_list=[[str(i)+'$_points',0],[str(i)+'$_points','Length']] )
`

**Reply by SebBerner** [reply]
I'm sorry, I forgot the operator. Intention was to test out whether the parameter can be used to change the number of traces. please use the corrected line:

`edb = Edb("my_project.aedb", edbversion="2022.2") 
edb.stackup.add_layer(layer_name="Top", material="copper", thickness="35um", layer_type="signal")

edb["$_points"] = 2

edb.add_design_variable(variable_name = "Length", variable_value = 4, is_parameter=False)

for i in range (0,int(edb["$_points"].value)):   
    edb.modeler.create_trace(layer_name='Top', width = 0.1, path_list=[[str(i)+'*$_points',0],[str(i)+'*$_points','Length']] )

edb.save_edb()
edb.close_edb()`

**Reply by hui-zhou-a** [reply]
I still don't understand why you put "$_points" in path_list. Can you describe what you expect, and what the code produces?

Will you get what you expect if "_points" is removed?
`edb.modeler.create_trace(layer_name='Top', width = 0.1, path_list=[[str(i),0],[str(i),'Length']] )`

**Reply by SebBerner** [reply]
The intention was to check whether the project variable works and effectively changes the x-position in EDT and the number of traces in EDT. What works is changing the position in EDT but not the number of traces:
With the imported EDB: 
![image](https://github.com/ansys/pyaedt/assets/155653845/91f9083e-ae68-4a7f-b891-230ef6cb69e8)
with chaning the project variable $_points to 3:
![image](https://github.com/ansys/pyaedt/assets/155653845/246eb034-531a-4120-bcfb-3e8b12f9a8a0)
This changes the x-position but doesn't change the number of traces. What I would like to have is 3 traces when I change $_points to 3 in EDT. For sure changing works when I do it in python but for my workflow I would like to make use of the project variable also in the GUI in EDT.

**Reply by hui-zhou-a** [reply]
@SebBerner $_points is a number in GUI. It cannot make an object of the layout disappear. I can not come up a solution for what you are aiming for.

**Reply by SebBerner** [reply]
Thanks for the clarification!
